File: app/src/main/java/com/bebediary/calendar/decorators/OneDayDecorator.java
package com.bebediary.calendar.decorators;

import android.content.Context;
import android.graphics.Color;
import android.graphics.Typeface;
import android.support.v4.app.FragmentActivity;
import android.text.style.ForegroundColorSpan;
import android.text.style.RelativeSizeSpan;
import android.text.style.StyleSpan;
import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;
import com.prolificinteractive.materialcalendarview.MaterialCalendarView;

import java.util.Date;

/**
 * Decorate a day by making the text big and bold
 */
public class OneDayDecorator implements DayViewDecorator {

    private CalendarDay date;
    private Context context;

    public OneDayDecorator(FragmentActivity activity) {
        date = CalendarDay.today();
        context = activity;
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        return date != null && day.equals(date);
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.addSpan(new StyleSpan(Typeface.BOLD));
        view.addSpan(new RelativeSizeSpan(1.1f));
        view.addSpan(new ForegroundColorSpan(Color.WHITE));
    }

    /**
     * We're changing the internals, so make sure to call {@linkplain MaterialCalendarView#invalidateDecorators()}
     */
    public void setDate(Date date) {
        this.date = CalendarDay.from(date);
    }
}

File: app/src/main/java/com/bebediary/calendar/decorator/OneDayDecorator.kt
package com.bebediary.calendar.decorator

import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.graphics.drawable.ColorDrawable
import android.text.style.ForegroundColorSpan
import android.text.style.RelativeSizeSpan
import android.text.style.StyleSpan
import androidx.core.content.res.ResourcesCompat
import com.bebediary.R
import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade

/**
 * 오늘일때 배경 설정 및
 * 글씨 두깨 변경
 */
class OneDayDecorator(
        private val context: Context
) : DayViewDecorator {

    // Background Color
    private val backgroundDrawable by lazy {
        ColorDrawable(ResourcesCompat.getColor(context.resources, R.color.colorPrimary, context.theme))
    }

    override fun shouldDecorate(day: CalendarDay): Boolean {
        return CalendarDay.today() == day
    }

    override fun decorate(view: DayViewFacade) {
        view.addSpan(StyleSpan(Typeface.BOLD))
        view.addSpan(RelativeSizeSpan(1.1f))
        view.addSpan(ForegroundColorSpan(Color.WHITE))
        view.setBackgroundDrawable(backgroundDrawable)
    }
}
